### PR TITLE
[ros_speech_recognition] Enable multi channel audio recognition

### PR DESCRIPTION
**Pull request by 708yamaguchi**
In this pull request, I enable multi channel audio recognition.

if audio topic's channel >= 2, single channel data is taken out from multi channel data.
This feature is based on `respeaker_node.py`, in which 1 channel audio data is taken out from 4 channel respeaker audio.
https://github.com/jsk-ros-pkg/jsk_3rdparty/blob/75e5e7ed0f650bffd1d434e522a72466be9735f7/respeaker_ros/scripts/respeaker_node.py#L278-L280

I was careful not to use numpy, which is additional dependency.

### CODE REVIEW

**Comment by 708yamaguchi** [comment]
I tested this feature with ThinkPad T460s built-in microphone (2 channels) and USB microphone (1 channel)

File: ros_speech_recognition/README.md [hunk 1]
```diff
@@ -96,14 +104,26 @@ This package uses Python package [SpeechRecognition](https://pypi.python.org/pyp
...
 
+* `~audio_topic` (`String`, default: `audio`)
+
+  Topic name of input audio data
+  
 * `~depth` (`Int`, default: `16`)
 
   Depth of audio signal
   
+* `~channels` (`Int`, default: `1`)
```
Comment (knorth55) [inline ros_speech_recognition/README.md:115]: does this mean number of channel?
or id of channel?

Comment (708yamaguchi) [reply ros_speech_recognition/README.md:115]: Thank you for your review!

This rosparam means the total number of channels in audio channels.
I modified the description.

Comment (knorth55) [reply ros_speech_recognition/README.md:115]: in that case, maybe it is better to use `n_channel` or something for param name.

Comment (708yamaguchi) [reply ros_speech_recognition/README.md]: OK.

I replace `channels` param with `n_channels`.

**Comment by 708yamaguchi** [comment]
The python built-in `array` module internally calls C language, but the number of bytes in an integer variable (e.g. `sizeof(short)`) depends on the CPU platform.
https://docs.python.org/ja/2.7/library/array.html

Therefore, I create dictionary variable to store the number of bytes in an integer variable.
```python
self.type_code = {}
for code in ['b', 'h', 'i', 'l']:
    self.type_code[array.array(code).itemsize] = code

...

dtype = self.type_code[self.depth/8]

**Pull request merged**